index.bs

```diff
@@ -94,8 +94,10 @@ interface Socket {
   readonly attribute ReadableStream readable;
   readonly attribute WritableStream writable;
 
+  readonly attribute Promise&lt;undefined> ready;
```

Comment:
Hate to bikeshed, but I think `opened` flows better with the existing `closed` nomenclature :)

Reply:
I'm also trying to think if theres a better way to represent this. Maybe by introducing an actual state machine pattern here.

Reply:
Hm, maybe, can you elaborate on what you have in mind?

Reply:
Something I've always pondered is explicit state. We could add a property such as `state` and it would essentially be a string enum `opened | connected | closed | destroyed`. It would probably take a lot of liberties from a proper state machine pattern such as the one Xstate employs: https://stately.ai/docs/xstate/basics/what-is-a-statechart

this is probably radically out-of-scope. i'm content with landing a `opened` promise and moving on lol. But maybe also we make Socket an EventTarget?

Reply:
I'm good with `opened`. The reason I went with `ready` here is that it is conceivable that an implementation might treat "ready to to be used" as a separate state than "the connection is actually established". Most of the time they would be one and the same but, for instance, in a 0RTT type of case (think something like QUIC that allows session resumption and allows a peer to begin optimistically transmitting early data during the TLS handshake) then the `ready` promise may be resolved earlier than the actual completion of the handshake. This, btw, is one of the reasons I avoided `opened` or `ready` in the first place, as it makes optimizations around that 0/0.5/1RTT case a bit more difficult.

Reply:
@Ethan-Arrowood ... I'm not convinced exposing a more detailed set of states is beneficial in the long run and I really don't like the idea of make `Socket` and `EventTarget`. I've always disliked the fact that `net.Socket` in Node.js is an `EventEmitter` in that I think it's entirely the wrong model to use.

Reply:
Ok, I've renamed `ready` to `opened`